test_othermlpackages/test_xgboost.py

```diff
@@ -1,11 +1,12 @@
 # Copyright (c) Microsoft Corporation and Fairlearn contributors.
 # Licensed under the MIT License.
 
+import pytest
 from . import package_test_common as ptc
 
 from fairlearn.reductions import DemographicParity
 
-import xgboost as xgb
+xgb = pytest.importorskip("xgboost")
```

Comment:
I'm a little confused as to why this is helping, because the test will fail if we can't import, won't it?

Reply:
the test will be skipped, not failed. This raises an exception caught by pytest which leads to it skipping the test rather than failing.

Reply:
It's related to why we have `test_othermlpackages` as a whole other directory.

Reply:
That being a separate directory doesn't make tools such as the ones in vscode to ignore them. This makes those tools to actually successfully "discover" the tests.

Reply:
I see. I think this goes back to our old problem of not wanting to have tests that are skipped without us noticing. Perhaps we were a little paranoid there (?)

If I understand correctly this would allow us to have these tests inside the normal `test` directory, right?

Reply:
... regardless of the answer I don't expect that to happen here 😆

Reply:
yes, they can go to the normal test suite if you want. AFAIK you can pass `-v` to `pytest` to get a list of skipped tests, if you wanna see what's skipped. Skipping is different than XFAIL though, that I'd understand more if you wanna make sure they pass sometimes?